packages/adblocker-extended-selectors/test/parse.test.ts

```diff
@@ -0,0 +1,700 @@
+            },
+            {
+              'argument': 'strong',
+              'content': ':has(strong)',
+              'length': 12,
+              'name': 'has',
+              'pos': [76, 88],
+              'subtree': {
+                'content': 'strong',
+                'length': 6,
+                'name': 'strong',
+                'pos': [0, 6],
+                'type': 'type',
+                'namespace': undefined,
+              },
+              'type': 'pseudo-class',
+            },
+            {
+              'argument': 'span > font > strong',
+              'content': ':has(span > font > strong)',
+              'length': 26,
+              'name': 'has',
+              'pos': [88, 114],
+              'subtree': {
+                'combinator': '>',
+                'left': {
+                  'combinator': '>',
+                  'left': {
+                    'content': 'span',
+                    'length': 4,
+                    'name': 'span',
+                    'pos': [0, 4],
+                    'type': 'type',
+                    'namespace': undefined,
+                  },
+                  'right': {
+                    'content': 'font',
+                    'length': 4,
+                    'name': 'font',
+                    'pos': [7, 11],
+                    'type': 'type',
+                    'namespace': undefined,
+                  },
+                  'type': 'complex',
+                },
+                'right': {
+                  'content': 'strong',
+                  'length': 6,
+                  'name': 'strong',
+                  'pos': [14, 20],
+                  'type': 'type',
+                  'namespace': undefined,
+                },
+                'type': 'complex',
+              },
+              'type': 'pseudo-class',
+            },
+          ],
+          'type': 'compound',
+        },
+        'type': 'complex',
+      });
+
+      expect(parse('.cls:has([href*="?utm_source="])')).to.eql({
+        'compound': [
+          {
+            'content': '.cls',
+            'length': 4,
+            'name': 'cls',
+            'pos': [0, 4],
+            'type': 'class',
+          },
+          {
+            'argument': '[href*="?utm_source="]',
+            'content': ':has([href*="?utm_source="])',
+            'length': 28,
+            'name': 'has',
+            'pos': [4, 32],
+            'subtree': {
+              'content': '[href*="?utm_source="]',
+              'length': 22,
+              'name': 'href',
+              'operator': '*=',
+              'pos': [0, 22],
+              'type': 'attribute',
+              'namespace': undefined,
+              'caseSensitive': undefined,
+              'value': '"?utm_source="',
+            },
+            'type': 'pseudo-class',
+          },
+        ],
+        'type': 'compound',
+      });
+
+      expect(parse('.cls > a > div[class^="foo"]:has(button[class^="bar"])')).to.eql({
```

Comment:
Errors from those expectations may not be very helpful as those would just asset the objects are different while you are interested in the exact difference. 
It may be worth to construct a smaller DOM to illustrate individual features in simpler "environment".